WcaOnRails/app/webpacker/components/CompetitionOverview/CompetitionTable.js

```diff
@@ -0,0 +1,178 @@
+import React from 'react';
+import {
+  List, Icon, Popup, Loader,
+} from 'semantic-ui-react';
+
+import I18n from '../../lib/i18n';
+import { dayDifferenceFromToday, PseudoLinkMarkdown } from '../../lib/utils/competition-table';
+
+function CompetitionTable({
+  competitionData,
+  title,
+  shouldShowRegStatus,
+  shouldIncludeCancelled,
+  selectedEvents,
+  isLoading,
+  hasMoreCompsToLoad,
+  isSortedByAnnouncement = false,
+  isRenderedAboveAnotherTable = false,
```

Comment:
Agree with the naming. I don't feel that it's necessary to pass the component in as a child when it can be defined in the same file, and perhaps move the conditions into the function for that component.

Reply:
Resolved because of footer component.